pkg/sink/kafka.go

```diff
@@ -88,26 +96,34 @@ func (s *KafkaSink) batchPublish(ctx context.Context, topic string, envelopes []
 			Headers: headers,
 		}
 		wg.Add(1)
+		var produceErr error // FIXME! Could probably do this similarly to pubsub sink and use a chan?
 		s.client.Produce(ctx, record, func(r *kgo.Record, err error) {
 			defer wg.Done()
 			if err != nil {
-				log.Error().Stack().Err(err).Msg("could not publish event to kafka")
+				log.Error().Err(err).Msg("could not publish record")
+				produceErr = err
 			} else {
 				offset := strconv.FormatInt(r.Offset, 10)
 				partition := strconv.FormatInt(int64(r.Partition), 10)
 				log.Trace().Msg("published event " + offset + " to topic " + topic + " partition " + partition)
 			}
 		})
+		if produceErr != nil {
```

Comment:
This is a race -- the Produce may finish concurrently with this read (or finish after, most likely)